src/main/scala/MilvusDataReader.scala

```diff
@@ -15,18 +15,84 @@ object MilvusDataReader {
       "s3a://"
     )
 
-    val insertDF = spark.read
-      .format("milvus")
-      .options(config.options)
-      .option(
-        MilvusOption.S3FileSystemTypeName,
-        fileSystemType
+    // Detect storage versions by querying segment info
+    val client = MilvusClient(
+      MilvusConnectionParams(
+        uri = config.uri,
+        token = config.token,
+        databaseName = config.options.getOrElse(MilvusOption.MilvusDatabaseName, "")
       )
-      .option(MilvusOption.ReaderType, "insert")
-      .option(MilvusOption.MilvusUri, config.uri)
-      .option(MilvusOption.MilvusToken, config.token)
-      .option(MilvusOption.MilvusCollectionName, config.collectionName)
-      .load()
+    )
```

Comment:
Why we have these logic here. This class is just a sugar, it should not knwo anything about Milvus' internall concept like V1, V2. Let's hid this in a depper layer